src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -0,0 +1,226 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $namespaces = array_map(function (NamespaceAnalysis $info) {
+            return $info->getFullName();
+        }, (new NamespacesAnalyzer())->getDeclarations($tokens));
+        $useMap = array_map(function (NamespaceUseAnalysis $info) {
+            return $info->getFullName();
+        }, (new NamespaceUsesAnalyzer())->getDeclarationsFromTokens($tokens));
+
+        if (!count($namespaces) && !count($useMap)) {
+            return;
+        }
+
+        $lastIndex = $tokens->count() - 1;
+        for ($index = $lastIndex; $index >= 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            // Return types are only available since PHP 7.0
+            $this->fixFunctionReturnType($tokens, $index, $namespaces, $useMap);
+            $this->fixFunctionArguments($tokens, $index, $namespaces, $useMap);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     * @param array  $namespaces
+     * @param array  $useMap
+     */
+    private function fixFunctionArguments(Tokens $tokens, $index, array $namespaces, array $useMap)
+    {
+        $arguments = (new FunctionsAnalyzer())->getFunctionArguments($tokens, $index);
+
+        foreach ($arguments as $argument) {
+            if (!$argument->hasType()) {
+                continue;
+            }
+
+            $shortType = $this->detectShortType($argument->getType(), $namespaces, $useMap);
+            if ($shortType === $argument->getType()) {
+                continue;
+            }
+
+            $tokens->overrideRange(
+                $argument->getTypeIndexStart(),
+                $argument->getTypeIndexEnd(),
+                $this->generateTokensForShortType($shortType)
+            );
+        }
+    }
+
+    /**
+     * @param Tokens                $tokens
+     * @param int                   $index
+     * @param array<string, string> $namespaces a list of all FQN namespaces in the file with the short name as key
+     * @param array<string, string> $useMap     a list of all FQN use statements in the file with the short name as key
+     */
+    private function fixFunctionReturnType(Tokens $tokens, $index, array $namespaces, array $useMap)
+    {
+        if (PHP_VERSION_ID < 70000) {
+            return;
+        }
+
+        $returnType = (new FunctionsAnalyzer())->getFunctionReturnType($tokens, $index);
+        if (!$returnType) {
+            return;
+        }
+
+        $shortType = $this->detectShortType($returnType->getType(), $namespaces, $useMap);
+        if ($shortType === $returnType->getType()) {
+            return;
+        }
+
+        $tokens->overrideRange(
+            $returnType->getStartIndex(),
+            $returnType->getEndIndex(),
+            $this->generateTokensForShortType($shortType)
+        );
```

Comment:
Line 156 to 163 is duplicated to 124 - 134, right?

If so, decoupling method would be nice and more bug proof